lib/components/molecules/Typeahead/Typeahead.js

```diff
@@ -0,0 +1,198 @@
+// @flow
+/**
+ *
+ * Typeahead
+ *
+ */
+import React, { useState } from 'react';
+import styled from 'styled-components';
+import classNames from 'classnames';
+
+import Input from '../../atoms/Input';
+import Label from '../../atoms/Label';
+import List from '../List';
+import ListItem from './ListItem';
+import styles from './Typeahead.style';
+import type { Props } from './types';
+import { ENTER_KEY, UP_ARROW, DOWN_ARROW, ESCAPE, TAB } from './constants';
+
+const Typeahead = ({
+  id,
+  className,
+  searchValue,
+  label,
+  isFieldLabel,
+  results,
+  onChangeHandler,
+  getResults,
+  renderCustomResultsView,
+  getSelectedResultValue,
+  customFilter,
+  ...otherProps
+}: Props): React.Element<any> => {
+  const [updatedResults, setFilteredResults] = useState([]);
+  const [activeResult, setActiveResult] = useState(0);
+
+  const triggerGetResults = val => {
+    return val && val.trim().length >= 3;
+  };
+
+  /**
+   * Function to return filtered results
+   * @param {*} allResults
+   * @param {*} inputText
+   */
+  const getFilteredResults = (allResults, inputText) => {
+    const filteredResults =
+      allResults &&
+      allResults.filter(Result => Result.toLowerCase().indexOf(inputText.toLowerCase()) > -1);
+    return filteredResults;
+  };
+
+  /**
+   * Function to handle item selection
+   * @param {'*'} e
+   */
+  const handleClick = item => {
+    setActiveResult(0);
+    getSelectedResultValue(item);
+  };
+
+  /**
+   * Function to handle text change in input
+   * @param {*} e
+   */
+  const handleChange = e => {
+    if (onChangeHandler) {
+      onChangeHandler(e);
+    }
+    const { value: inputText } = e.target;
+    let filteredResults = results;
+    if (triggerGetResults(inputText)) {
+      if (getResults) {
+        getResults(inputText);
+      } else if (customFilter) {
+        filteredResults = customFilter(results, inputText);
+      } else {
+        filteredResults = getFilteredResults(results, inputText);
+      }
+      setFilteredResults(filteredResults);
```

Comment:
still duplicate calls to setFilteredResults, this could be just one call outside if else